### PR TITLE
Enhance documentation on automated testing

### PR DESCRIPTION
**Pull request by sukidog**

### CODE REVIEW

File: docs/AUTOMATED_TESTS.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Tests are grouped into two categories
+
+* Tests that validate how DAPLink is used for programming the target device and even the HIC itself via MSD--by copying an image file to the MSD drive (most users will drag-and-drop). These tests do not run by default but can be turned on by specifying --testdl.
+
+
+* Tests that validate the MSD, CDC, and CMSIS-DAP endpoints. These tests run by default, but can be turned off by specifying --notestendpt
+
+The endpoint tests rely on a target application--a small UART program that implements a simple protocol. The target app is used by all three endpoint sub-tests.
+
+For mbed boards that are registered on mbed.org, the test builds the target application code on the fly in the cloud using the mbed RESTful Compile API. It then downloads the resulting image and downloads it to the target. The UART application source is in an mbed git repository (https://developer.mbed.org/users/c1728p9/code/daplink-validation/). For this workflow, you must specify your mbed.org user id and password via --user and --password.
```
Comment (c1728p9) [inline docs/AUTOMATED_TESTS.md]: Currently, the project is in a mercurial repository on mbed, not git.  What do you think about something like this:
```
The UART application source is the mbed mercurial repository [daplink-validation](https://developer.mbed.org/users/c1728p9/code/daplink-validation/)
```
Without the code formatting:
The UART application source is the mbed mercurial repository [daplink-validation](https://developer.mbed.org/users/c1728p9/code/daplink-validation/)

File: docs/PORT_BOARD.md [hunk 2]
```diff
@@ -46,4 +46,15 @@ This assumes there is already target support present in the codebase. If adding
...
\ No newline at end of file
+Now running `progen generate -t uvision` will create project files including the new board that can be developed and debugged. For more information about the yaml format [see the project_generator documentation.](https://github.com/project-generator/project_generator/wiki/Getting_started)
+
+### Adding Board to Automated Tests
+Update `test/into.py`. There are various dictionaries that you will need to add to
+* Add the name of the interface (as specified in the projects.yaml) to FIRMWARE_NAME_TO_BOARD_ID and associate it with the board ID
+
+* Add to TARGET_NAME_TO_BOARD_ID. What you use for the key will depend on whether your board is mbed enabled _and_ registered with mbed.org.
+	* If it is, the automated tests will build the target UART application on the fly using the RESTful Compile API. The source code is in an mbed.org git repository, and it will build the application in the cloud, download it to the PC, then download it to the target.
```
Comment (c1728p9) [inline docs/PORT_BOARD.md]: mercurial instead of git again

**Comment by c1728p9** [comment]
Thanks for the updated docs.  Info about testing and adding new targets to the test framework was missing, so the PR is really appreciated.

**Comment by sukidog** [comment]
You bet. I made adjustments to address your feedback. Also, I realized my PORT_BOARD section lacked a key detail for each case, so a bit of churn there.

**Comment by c1728p9** [comment]
Looks good! Thanks

**Pull request merged**